### PR TITLE
RT-563-Color library

### PR DESCRIPTION
**Pull request by yashpmohod**
## Overview
use tinycolor2 library.

## How it was tested
Ran showcase app locally.

## Checklist

- [x] The title contains a short meaningful summary
- [x] I have added a link to this PR in the Jira issue
- [x] I have described how this was tested
- [ ] I have included screen shots for changes that affect the user interface
- [x] I have updated unit tests
- [x] I have run unit tests locally
- [ ] I have updated documentation (including README)

### CODE REVIEW

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts [hunk 1]
```diff
@@ -22,22 +24,16 @@ export class DataLayerColorService {
...
-  }
-
   addTransparency(color: string | undefined): string | undefined {
-    if (typeof color === 'string' && !color.endsWith('33')) {
-      return color + '33';
-    }
-    return color;
+    const newcolor = tinycolor(color);  
+    newcolor.setAlpha(.5);
+    return newcolor.toString();
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:30]: `.5` is much darker than hex `33`. it's hard to see the lines when the annotation is so dark.

```suggestion
    const newcolor = tinycolor(color);  
    newcolor.setAlpha(.2);
    return newcolor.toString();
```

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts [hunk 2]
```diff
@@ -22,22 +24,16 @@ export class DataLayerColorService {
...
   }
 
   setColor(dataset: Dataset, color: string): void {
     const line = dataset as Dataset<'line'>;
     line.borderColor = color;
-    line.backgroundColor = color + '33'; // temporary dirty hack to set opacity. assumes color is in 6-character hex format.
+    line.backgroundColor = this.addTransparency(color); // temporary dirty hack to set opacity. assumes color is in 6-character hex format.
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:36]: please remove the comment because it no longer applies

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts [hunk 3]
```diff
@@ -50,7 +46,7 @@ export class DataLayerColorService {
     return undefined;
   }
 
-  /** build a CSS linear gradient that includes colors from all datasets in the layer */
+  //  build a CSS linear gradient that includes colors from all datasets in the layer
   getColorGradient(layer: DataLayer) {
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:50]: why did you change this from a doc comment to a line comment?

**Review by semicolin — CHANGES_REQUESTED** [review]

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts (hunk 3 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:50]: Okay, I will change the line comment to a doc comment.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts (hunk 1 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:30]: Yes, I will change it as you suggested.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts (hunk 2 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:36]: Okay, I will remove the comments.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts [hunk 4]
```diff
@@ -50,7 +46,7 @@ export class DataLayerColorService {
     return undefined;
   }
 
-  /** build a CSS linear gradient that includes colors from all datasets in the layer */
+  /* build a CSS linear gradient that includes colors from all datasets in the layer */
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts:49]: The extra `*` in `/**` means this is a documentation comment. Tools like jsdoc/typedoc that generate API documentation from the source code will look for it. It should not be removed.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts [hunk 5]
```diff
@@ -6,9 +6,9 @@ import { DataLayerMergeService } from './data-layer-merge.service';
 import produce, { castDraft } from 'immer';
 import { DataLayerModule } from './data-layer.module';
 import { zip } from 'lodash-es';
-import { NumberRange } from '../utils';
 
-/** A service for asynchronously retrieving DataLayers */
+
+/* A service for asynchronously retrieving DataLayers */
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts:11]: ...same here

**Review by semicolin — CHANGES_REQUESTED** [review]
FYI: https://typedoc.org/guides/doccomments/

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts (hunk 5 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts:11]: okay, I will update this.

**Review by semicolin — APPROVED** [review]
